plugins/inputs/ipmi/ipmi.go

```diff
@@ -0,0 +1,113 @@
+// ipmi
+package ipmi
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Ipmi struct {
+	Servers []string
+	runner  Runner
+}
+
+var sampleConfig = `
+  ## specify servers via a url matching:
+  ##  [username[:password]@][protocol[(address)]]
+  ##  e.g.
+  ##    root:passwd@lan(127.0.0.1)
+  ##
+  servers = ["USERID:PASSW0RD@lan(192.168.1.1)"]
+`
+
+func NewIpmi() *Ipmi {
+	return &Ipmi{
+		runner: CommandRunner{},
+	}
+}
+
+func (m *Ipmi) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *Ipmi) Description() string {
+	return "Read metrics from one or many bare metal servers"
+}
+
+func (m *Ipmi) Gather(acc telegraf.Accumulator) error {
+	if m.runner == nil {
+		m.runner = CommandRunner{}
+	}
+	for _, serv := range m.Servers {
+		err := m.gatherServer(serv, acc)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Ipmi) gatherServer(serv string, acc telegraf.Accumulator) error {
+	conn := NewConnection(serv)
+
+	res, err := m.runner.Run(conn, "sdr")
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(res, "\n")
+
+	for i := 0; i < len(lines); i++ {
+		vals := strings.Split(lines[i], "|")
+		if len(vals) == 3 {
+			tags := map[string]string{"host": conn.Hostname, "inst": trim(vals[0])}
```

Comment:
The default host tag will use the local hostname which the telegraf agent running on,but ipmi plugin collect remote bare metal host's metric.
I think the local hostname is not usable,but a more tag will increase the series cardinality.